lib/active_merchant/billing/gateways/monei.rb

```diff
@@ -0,0 +1,314 @@
+          end unless money.nil?
+        end
+      end
+
+      # Private: Add account part to XML
+      def add_account(xml, credit_card)
+        xml.Account do
+          xml.Holder credit_card.name
+          xml.Number credit_card.number
+          xml.Brand credit_card.brand.upcase
+          xml.Expiry(:month => credit_card.month, :year => credit_card.year)
+          xml.Verification credit_card.verification_value
+        end
+      end
+
+      # Private: Add customer part to XML
+      def add_customer(xml, credit_card, options)
+        requires!(options, :billing_address)
+        address = options[:billing_address]
+        xml.Customer do
+
+          xml.Name do
+            xml.Given options[:given] || credit_card.first_name
+            xml.Family options[:family] || credit_card.last_name
+          end
+          xml.Address do
+            xml.Street address[:address1].to_s
+            xml.Zip address[:zip].to_s
+            xml.City address[:city].to_s
+            xml.State address[:state] if address.has_key? :state
+            xml.Country address[:country].to_s
+          end
+          xml.Contact do
+            xml.Email options[:email] || 'noemail@monei.net'
+            xml.Ip options[:ip] || '0.0.0.0'
+          end
+
+        end
+      end
+
+      # Private: Parse XML response from Monei servers
+      def parse(body)
+        xml = Nokogiri::XML(body)
+
+        {
+            :unique_id => xml.xpath("//Response/Transaction/Identification/UniqueID").text,
+            :status => (tr_status_code xml.xpath("//Response/Transaction/Processing/Status/@code").text),
+            :reason => (tr_status_code xml.xpath("//Response/Transaction/Processing/Reason/@code").text),
+            :message => xml.xpath("//Response/Transaction/Processing/Return").text
+        }
+      end
+
+      # Private: Send XML transaction to Monei servers and create AM response
+      def commit(xml)
+        url = (test? ? test_url : live_url)
+
+        response = parse(ssl_post(url, post_data(xml), 'Content-Type' => 'application/x-www-form-urlencoded;charset=UTF-8'))
+
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: authorization_from(response),
+          test: test?
+        )
+      end
+
+      # Private: Decide success from servers response
+      def success_from(response)
+        response[:status] == :success or response[:status] == :new
+      end
+
+      # Private: Get message from servers response
+      def message_from(response)
+        response[:message]
+      end
+
+      # Private: Get authorization code from servers response
+      def authorization_from(response)
+        response[:unique_id]
+      end
+
+      # Private: Encode POST parameters
+      def post_data(xml)
+        "load=#{CGI.escape(xml)}"
+      end
+
+      # Private: Translate Monei status code to native ruby symbols
+      def tr_status_code(code)
```

Comment:
@j-mutter I could translate all the rejections into `:card_declined` but the other values are more status that error codes.